apps/api/src/questions/questions.service.ts

```diff
@@ -171,36 +98,32 @@ export class QuestionsService {
     }
 
     if (answerQuestion.answer.length < 1)
-      return await this.upsertAnswer(
-        trx,
+      return await this.questionsRepository.upsertAnswer(
         answerQuestion.courseId,
         questionData.lessonId,
         questionData.questionId,
         userId,
         lastAnswerId,
         [],
+        trx,
       );
 
-    const answers: { answer: string }[] = await trx
-      .select({
-        answer: questionAnswerOptions.optionText,
-      })
-      .from(questionAnswerOptions)
-      .where(
-        and(
-          eq(questionAnswerOptions.questionId, questionData.questionId),
-          inArray(
-            questionAnswerOptions.id,
-            answerQuestion.answer.map((a) => {
-              if (typeof a !== "string") {
-                return a.value;
-              }
-
-              return a;
-            }),
-          ),
-        ),
-      );
+    const answerList = answerQuestion.answer.map((a) => {
+      if (typeof a !== "string") {
+        return a.value;
+      }
+
+      return a;
+    });
+
+    if (!answerList || answerList.length === 0)
```

Comment:
You can use here 
`if (!answerList?.length) {
  throw new NotFoundException("User answers not found");
}`